### ISSUE TITLE
bug(Worker): Not loading latest version of specified script

### ISSUE THREAD

**Issue by wongjiahau** [denoland/deno#6116]
## Steps to reproduce
1. Create a file `script.ts` with the content `console.log('hello')`.
2. Create another file `entry.ts` with the following content:
```ts
while(true) {
  const worker = new Worker('./script.ts', {type: 'module'})
  await new Promise(resolve => setTimeout(resolve, 1000))
  worker.terminate()
}
```
3. Run this command: `deno run --allow-read main.ts`.
4. Modify the content of `script.ts` to `console.log('bye')` and save it.

## Expected result
The subsequent log of `deno run --allow-read main.ts` should be `bye`.

## Actual result
After step 4, the subsequent log of  `deno run --allow-read main.ts` is still `hello`.

## Environment:
```
macOS Catalina (v10.15.4)
```
```
$ deno --version
deno 1.0.5
v8 8.4.300
typescript 3.9.2
```

## Is this introduced lately?
No, this is happening on every version of Deno since `v1.0.0` until `v1.0.5`.

## Current workaround
Refer https://stackoverflow.com/a/41790024/6587634

**Comment by ry** [comment]
Thanks for the report. I guess it's related to our module cache - it should be bypassed for dynamic loads (Worker constructors and dynamic imports).

@bartlomieju what do you think?

**Comment by bartlomieju** [comment]
Yes, this is the same problem as in #5631 and #4743

**Comment by kitsonk** [comment]
While it hasn't fixed #5631 and #4743, I believe the compiler refactor has fixed this one, but I need to create a test to validate that.

**Comment by ry** [comment]
I just tried the reproduction with 1.5.2 and it works as expected.

**Comment by ry** [comment]
Sorry - I didn't understand the example correctly. This is still a bug.

**Comment by dev-nicolaos** [comment]
This bug appears to have come back. I setup the steps to reproduce today running Deno v1.27.0 on Windows and did not get the expected result. After changing the text in _script.ts_ and saving the file Deno continued to print the original string. I also tried closing the worker from itself (`self.close`) rather than using `worker.terminate` but got the same unexpected behavior.

**Comment by bartlomieju** [comment]
@dev-nicolaos can you please open a new issue?

**Comment by dev-nicolaos** [comment]
@bartlomieju I've opened #16503
